stablehlo/dialect/TypeInference.cpp

```diff
@@ -71,16 +71,37 @@ namespace {
 //===----------------------------------------------------------------------===//
 template <typename T>
 bool allQuantized(ArrayRef<Type> typeRange) {
-  return llvm::all_of(typeRange, [&](Type val) {
-    return val.cast<ShapedType>().getElementType().isa<T>();
-  });
+  return llvm::all_of(
+      typeRange, [&](Type val) { return getElementTypeOrSelf(val).isa<T>(); });
 }
 
 template <typename T>
 bool noneQuantized(ArrayRef<Type> typeRange) {
-  return llvm::all_of(typeRange, [&](Type val) {
-    return !val.cast<ShapedType>().getElementType().isa<T>();
-  });
+  return llvm::all_of(
+      typeRange, [&](Type val) { return !getElementTypeOrSelf(val).isa<T>(); });
+}
+
+// Try to match scale and zero_point for per_tensor quantized inputs
```

Comment:
Try to make this generic for per-tensor and per-axis.. The name of the function is sufficient to assume that it can be applied for both the cases.

Reply:
refactored isSameQuantScaleZeroPoint to include per-axis Q support, removed location logging, removed per scale or per zero_point mismatch logging. Now caller will log the error.  We won't get scale mismatch or zero_point mismatch granularity in error message instead generic error message "expect same quantization scale and zero_point"